src/components/capture/creation/SpecForm.tsx

```diff
@@ -29,45 +29,50 @@ const defaultAjv = createAjv({ useDefaults: true });
 function NewCaptureSpecForm(props: NewCaptureSpecFormProps) {
     const { state, dispatch, endpoint, readonly, displayValidation } = props;
 
+    const { isIdle, isLoading, isSuccess, error, data } =
+        useConnectorImageSpec(endpoint);
+
     const {
-        loading,
-        error,
-        data: {
-            specSchema,
-            links: { discovery, documentation },
-        },
-    } = useConnectorImageSpec(endpoint);
+        links: { discovery, documentation },
+        endpointSchema,
+    } = data;
```

Comment:
This hook provides a default state. Makes it nice to destructure.... but does mean down below we need some more specific tests.